### PR TITLE
Code Clean up: Convert to null-coalescing operator ??

### PR DESCRIPTION
**Pull request by BaronGreenback**
As suggested by ReSharper.

Changed 
```
if (varname == null)
{
  varname = ....
}
```
to `varname ?? ....` and `varname ??=...` (where appropriate)

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 1]
```diff
@@ -1187,43 +1187,28 @@ private void SetAudioInfoFromTags(MediaInfo audio, Dictionary<string, string> ta
             FetchStudios(audio, tags, "label");
 
             // These support mulitple values, but for now we only store the first.
-            var mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Artist Id"));
-            if (mb == null)
-            {
-                mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMARTISTID"));
-            }
+            var mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Artist Id")) ??
+                        GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMARTISTID"));
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1191]:
```suggestion
            var mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Artist Id"))
                       ?? GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMARTISTID"));
```

I think the general rule is that if the operation spans multiple lines, the operator is on the next line

File: Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs [hunk 2]
```diff
@@ -2239,14 +2239,10 @@ private void UpdateTimersForSeriesTimer(SeriesTimerInfo seriesTimer, bool update
...
-                if (existingTimer == null)
-                {
-                    existingTimer = string.IsNullOrWhiteSpace(timer.ProgramId)
-                        ? null
-                        : _timerProvider.GetTimerByProgramId(timer.ProgramId);
-                }
+                var existingTimer = _timerProvider.GetTimer(timer.Id) ??
+                                    (string.IsNullOrWhiteSpace(timer.ProgramId)
+                                        ? null
+                                        : _timerProvider.GetTimerByProgramId(timer.ProgramId));
```
Comment (crobibero) [inline Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs:2245]: same

File: Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs [hunk 3]
```diff
@@ -787,14 +787,11 @@ public async Task<List<ChannelInfo>> GetChannels(ListingsProviderInfo info, Canc
...
-                    station = new ScheduleDirect.Station
-                    {
-                        stationID = channel.stationID
-                    };
-                }
+                var station = allStations.Find(item => string.Equals(item.stationID, channel.stationID, StringComparison.OrdinalIgnoreCase)) ??
+                                new ScheduleDirect.Station
+                                {
+                                    stationID = channel.stationID
+                                };
```
Comment (crobibero) [inline Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs:794]: same

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 4]
```diff
@@ -1187,43 +1187,28 @@ private void SetAudioInfoFromTags(MediaInfo audio, Dictionary<string, string> ta
...
 
             audio.SetProviderId(MetadataProvider.MusicBrainzAlbumArtist, mb);
 
-            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Artist Id"));
-            if (mb == null)
-            {
-                mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ARTISTID"));
-            }
+            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Artist Id")) ??
+                    GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ARTISTID"));
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1196]:
```suggestion
            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Artist Id"))
                   ?? GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ARTISTID"));
```

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 5]
```diff
@@ -1187,43 +1187,28 @@ private void SetAudioInfoFromTags(MediaInfo audio, Dictionary<string, string> ta
...
 
             audio.SetProviderId(MetadataProvider.MusicBrainzArtist, mb);
 
-            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Id"));
-            if (mb == null)
-            {
-                mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMID"));
-            }
+            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Id")) ??
+                    GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMID"));
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1201]:
```suggestion
            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Album Id"))
                   ?? GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_ALBUMID"));
```

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 6]
```diff
@@ -1187,43 +1187,28 @@ private void SetAudioInfoFromTags(MediaInfo audio, Dictionary<string, string> ta
...
 
             audio.SetProviderId(MetadataProvider.MusicBrainzAlbum, mb);
 
-            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Group Id"));
-            if (mb == null)
-            {
-                mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASEGROUPID"));
-            }
+            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Group Id")) ??
+                    GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASEGROUPID"));
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1206]:
```suggestion
            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Group Id"))
                   ?? GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASEGROUPID"));
```

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 7]
```diff
@@ -1187,43 +1187,28 @@ private void SetAudioInfoFromTags(MediaInfo audio, Dictionary<string, string> ta
...
 
             audio.SetProviderId(MetadataProvider.MusicBrainzReleaseGroup, mb);
 
-            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Track Id"));
-            if (mb == null)
-            {
-                mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASETRACKID"));
-            }
+            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Track Id")) ??
+                    GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASETRACKID"));
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1211]:
```suggestion
            mb = GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MusicBrainz Release Track Id"))
                   ?? GetMultipleMusicBrainzId(FFProbeHelpers.GetDictionaryValue(tags, "MUSICBRAINZ_RELEASETRACKID"));
```

**Review by crobibero — COMMENTED** [review]
Missed some.
I may have the indentation incorrect as it's hard to tell on Github

**Review by Ullmie02 — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**